src/overlays/actors/ovl_Bg_Gnd_Iceblock/z_bg_gnd_iceblock.c

```diff
@@ -10,21 +10,23 @@
 
 #define THIS ((BgGndIceblock*)thisx)
 
-void BgGndIceblock_Init(Actor* thisx, GlobalContext* globalCtx);
-void BgGndIceblock_Destroy(Actor* thisx, GlobalContext* globalCtx);
-void BgGndIceblock_Update(Actor* thisx, GlobalContext* globalCtx);
-void BgGndIceblock_Draw(Actor* thisx, GlobalContext* globalCtx);
+typedef enum {
+    /* 0 */ GNDICE_IDLE,
+    /* 1 */ GNDICE_FALL,
+    /* 2 */ GNDICE_HOLE
+} BgGndIceblockAction;
 
-void func_8087A0C8(BgGndIceblock* this, GlobalContext* globalCtx);
-void func_8087A184(BgGndIceblock* this, GlobalContext* globalCtx);
-void func_8087A248(BgGndIceblock* this, GlobalContext* globalCtx);
-void func_8087A300(BgGndIceblock* this, GlobalContext* globalCtx);
-void func_8087A39C(BgGndIceblock* this, GlobalContext* globalCtx);
+void BgGndIceblock_Init(BgGndIceblock* this, GlobalContext* globalCtx);
+void BgGndIceblock_Destroy(BgGndIceblock* this, GlobalContext* globalCtx);
+void BgGndIceblock_Update(BgGndIceblock* this, GlobalContext* globalCtx);
+void BgGndIceblock_Draw(BgGndIceblock* this, GlobalContext* globalCtx);
```

Comment:
I think these functions should just take `Actor* thisx`?

Reply:
yes they should.